File: api/app/models/dee_jay.rb
class DeeJay < ApplicationRecord
  has_secure_password
  
  has_many :radio_shows
  has_many :songs, -> { distinct }, through: :radio_shows
  has_many :artists, -> { distinct }, through: :songs
  
  validates_presence_of :name, :email, :password_digest
  validates :email, uniqueness: true

  ROLE_ADMINISTRATOR = 1
  
  def is_admin?
    return self[:roles] & ROLE_ADMINISTRATOR
  end
end

File: client/src/app/models/dee-jay.ts
import { Artist } from "./artist";
import { RadioShow } from "./radio-show";
import { Song } from "./song";

export interface DeeJay {
  id: number,
  email: string,
  mixcloud_user: string,
  name: string,
  radio_shows: RadioShow[] | undefined,
  songs: Song[] | undefined,
  artists: Artist[] | undefined,
}